lib/matplotlib/figure.py

```diff
@@ -1113,9 +1113,29 @@ def legend(self, *args, **kwargs):
             # extra_args = extra_args[1:]
             pass
         transform = kwargs.pop('bbox_transform', self.transSubfigure)
+        outside = None
+        if 'loc' in kwargs:
+            # need to protect this in case loc was passed in
+            # positionally...
```

Comment:
I think loc cannot be positionally, because tih would then need to be in `extra_args`, but `loc` is kw-only in [Legend](https://matplotlib.org/stable/api/legend_api.html#matplotlib.legend.Legend), so it would fail a few lines down, when you pass `loc` positionally there via `*extra_args`.

Reply:
*loc* is not yet kw-only in `legend.__init__`.

Reply:
oh, I see. That will only become kwonly in 3.8.